Length of the field blocks differ in Structs field resulting in a java.lang.IllegalArgumentException

```
com.facebook.presto.spi.PrestoException: Failed to read ORC file:
	at com.facebook.presto.hive.orc.OrcSelectivePageSource.getNextPage(OrcSelectivePageSource.java:100)
	at com.facebook.presto.operator.ScanFilterAndProjectOperator.processPageSource(ScanFilterAndProjectOperator.java:272)
	at com.facebook.presto.operator.ScanFilterAndProjectOperator.getOutput(ScanFilterAndProjectOperator.java:237)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:379)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:283)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:675)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:276)

Caused by: java.lang.IllegalArgumentException: length of field blocks differ: field 0: 232, block 1: 34
	at com.facebook.presto.spi.block.RowBlock.validateConstructorArguments(RowBlock.java:93)
	at com.facebook.presto.spi.block.RowBlock.fromFieldBlocks(RowBlock.java:52)
	at com.facebook.presto.orc.reader.StructSelectiveStreamReader.getBlockView(StructSelectiveStreamReader.java:545)
	at com.facebook.presto.orc.reader.StructSelectiveStreamReader.getBlockView(StructSelectiveStreamReader.java:540)
	at com.facebook.presto.orc.reader.StructSelectiveStreamReader.getBlockView(StructSelectiveStreamReader.java:540)
```

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java
@@ -670,7 +670,7 @@ public class TestHivePushdownFilterQueries
     {
         List<String> types = ImmutableList.of(BOOLEAN, TINYINT, SMALLINT, INTEGER, BIGINT, DOUBLE, REAL, VARCHAR, DATE);
 
-        String query = String.format("SELECT orderkey, CAST(ROW(%s, 1) AS ROW(%s, a INTEGER)) as struct FROM orders",
+        String query = String.format("SELECT orderkey, CAST(ROW(%s, orderkey) AS ROW(%s, orderkey INTEGER)) as struct FROM orders",
                 types.stream()
                         .map(type -> "null")
                         .collect(joining(", ")),
@@ -684,9 +684,13 @@ public class TestHivePushdownFilterQueries
             for (String type : types) {
                 assertQuery(
                         format(
-                                "SELECT struct.a, struct.null_%s FROM test_all_nulls_in_struct WHERE struct IS NOT NULL AND orderkey %% 2 = 0",
+                                "SELECT struct.orderkey, struct.null_%s FROM test_all_nulls_in_struct WHERE struct IS NOT NULL AND orderkey %% 2 = 0",
                                 type.toLowerCase(getSession().getLocale())),
-                        "SELECT 1, null FROM orders WHERE orderkey % 2 = 0");
+                        "SELECT orderkey, null FROM orders WHERE orderkey % 2 = 0");
+
+                assertQuery(format("SELECT orderkey from test_all_nulls_in_struct WHERE struct.null_%s is NULL AND struct.orderkey > 3000 " +
+                                "AND length(CAST(struct.null_%s AS VARCHAR)) IS NULL", type.toLowerCase(getSession().getLocale()), type.toLowerCase(getSession().getLocale())),
+                        "SELECT orderkey FROM orders WHERE orderkey > 3000");
             }
         }
         finally {

--- a/presto-orc/src/main/java/com/facebook/presto/orc/reader/BooleanSelectiveStreamReader.java
+++ b/presto-orc/src/main/java/com/facebook/presto/orc/reader/BooleanSelectiveStreamReader.java
@@ -392,7 +392,7 @@ public class BooleanSelectiveStreamReader
         checkState(!valuesInUse, "BlockLease hasn't been closed yet");
 
         if (allNulls) {
-            return newLease(new RunLengthEncodedBlock(NULL_BLOCK, outputPositionCount));
+            return newLease(new RunLengthEncodedBlock(NULL_BLOCK, positionCount));
         }
 
         boolean includeNulls = nullsAllowed && presentStream != null;

--- a/presto-orc/src/main/java/com/facebook/presto/orc/reader/ByteSelectiveStreamReader.java
+++ b/presto-orc/src/main/java/com/facebook/presto/orc/reader/ByteSelectiveStreamReader.java
@@ -385,7 +385,7 @@ public class ByteSelectiveStreamReader
         checkState(!valuesInUse, "BlockLease hasn't been closed yet");
 
         if (allNulls) {
-            return newLease(new RunLengthEncodedBlock(NULL_BLOCK, outputPositionCount));
+            return newLease(new RunLengthEncodedBlock(NULL_BLOCK, positionCount));
         }
 
         boolean includeNulls = nullsAllowed && presentStream != null;

--- a/presto-orc/src/main/java/com/facebook/presto/orc/reader/DoubleSelectiveStreamReader.java
+++ b/presto-orc/src/main/java/com/facebook/presto/orc/reader/DoubleSelectiveStreamReader.java
@@ -388,7 +388,7 @@ public class DoubleSelectiveStreamReader
         checkState(!valuesInUse, "BlockLease hasn't been closed yet");
 
         if (allNulls) {
-            return newLease(new RunLengthEncodedBlock(NULL_BLOCK, outputPositionCount));
+            return newLease(new RunLengthEncodedBlock(NULL_BLOCK, positionCount));
         }
 
         boolean includeNulls = nullsAllowed && presentStream != null;

